src/wallet/db.cpp

```diff
@@ -402,11 +403,41 @@ CDB::CDB(CWalletDBWrapper& dbw, const char* pszMode, bool fFlushOnCloseIn) : pdb
                             0);
 
             if (ret != 0) {
+                error = strprintf("CDB: Error %d, can't open database %s", ret, strFilename);
+            } else {
+                // Make sure database has a unique fileid. If it doesn't, throw
+                // an error. BDB caches do not work properly when more than one
+                // open database has the same fileid (values written to one
+                // database may show up in reads to other databases).
+                //
+                // Bitcoin will never create different databases with the same
+                // fileid, but this error can be triggered if users manually
+                // copy database files.
+                u_int8_t fileid[DB_FILE_ID_LEN];
+                ret = pdb->get_mpf()->get_fileid(fileid);
+                if (ret != 0) {
+                    error = strprintf("CDB: Can't open database %s (get_fileid error %d)\n", strFilename, ret);
+                }
+
+                for (const auto& item : env->mapDb) {
+                    u_int8_t item_fileid[DB_FILE_ID_LEN];
+                    if (item.second && item.second->get_mpf()->get_fileid(item_fileid) == 0 &&
+                        memcmp(fileid, item_fileid, sizeof(fileid)) == 0) {
+                        const char* item_filename = nullptr;
+                        item.second->get_dbname(&item_filename, nullptr);
+                        error = strprintf("CDB: Can't open database %s (duplicates fileid %s from %s)", strFilename,
```

Comment:
This error overrides the previous.

Reply:
Should `+= std::endl +` instead?

Reply:
Fixed override in e32c2a51b757ab764ec5dbe8bf04b3fc4aa3cb5f, but I'm not clear on what std::endl comment is referring to.

Reply:
Nevermind, but the suggestion was to accumulate the errors.